### ISSUE TITLE
[Service Bus] setAutoDeleteOnIdle is not working for subscription a topic

### ISSUE THREAD

**Issue by jsquire** [Azure/azure-sdk-for-java#21724]
@johnwickfootball commented on [Fri May 21 2021](https://github.com/Azure/azure-sdk-for-net/issues/21276)

After sitting up setAutoDeleteOnIdle for 7mins the subscription is not getting delete. I am using Java with Azure service bus version 7.0.2

```
 SubscriptionDescription  sub = ManagementClient.createSubscription(topicName, subName);
 sub.setAutoDeleteOnIdle(duration(7mins);
 ManagementClient.updateSubcribeption(sub);
```

---

@jsquire commented on [Sat May 22 2021](https://github.com/Azure/azure-sdk-for-net/issues/21276#issuecomment-846413643)

Hi @johnwickfootball.  Thanks for reaching out and we regret that you're experiencing difficulties.   I'm going to transfer this over to the Java SDK repository, as the folks there would be better able to assist.

**Comment by joshfree** [comment]
@YijunXieMS could you help @johnwickfootball on this issue?

**Comment by ramya-rao-a** [comment]
@ki1729 Please see if you can reproduce this issue. If so, then we can work with the service team to see what is wrong. You can also check with our friends from other languages to see if this a problem with our assumption or the service itself

**Comment by ZejiaJiang** [comment]
Hi @ramya-rao-a ,

I'm from java sdk team. I tried to reproduce this bug in version 7.0.2 and latest version. And it works well. Have service team fixed this bug already?

**Comment by ramya-rao-a** [comment]
@ZejiaJiang Looks like you used the azure-messaging-servicebus package which does not have a class called `SubscriptionDescription`. This class is in the older azure-servicebus package. Can you try reproducing the issue in the older one? If it works there as well, then we can close the issue

**Comment by ZejiaJiang** [comment]
@ramya-rao-a ok, I'll try older version today.

I tried this version. Do you know the certain version?
```xml
<dependency>
        <groupId>com.azure</groupId>
         <artifactId>azure-messaging-servicebus</artifactId>
         <version>7.0.2</version>
</dependency>
```

**Comment by ramya-rao-a** [comment]
@ZejiaJiang, I meant an older SDK: https://mvnrepository.com/artifact/com.microsoft.azure/azure-servicebus

**Comment by ZejiaJiang** [comment]
> @ZejiaJiang, I meant an older SDK: https://mvnrepository.com/artifact/com.microsoft.azure/azure-servicebus

I got it. Thank you.

**Comment by ZejiaJiang** [comment]
Hi @ramya-rao-a ,

I checked the changelog of package `azure-messaging-servicebus` , only `azure-messaging-servicebus:7.0.0-beta.4` support `SubscriptionDescription` and `createSubscription` .I tested this package and it works.

And then I tested `azure-servicebus:3.6.6`   `azure-servicebus:3.6.0` and  `azure-servicebus:3.5.1`. All of them work as well.

Do I need more test of  `azure-servicebus` or we can close this issue?
